### ISSUE TITLE
Handle deprecated functions better in table of functions which can be called prior to initializing MPI

### ISSUE THREAD

**Issue by hppritcha** [mpi-forum/mpi-issues#504]
# Problem

In addressing issue #456, some discussion arose about handling the issue of deprecated functions and whether to include them in the table of pre-init functions.  Suggestions for improvement included adding deprecated functions to the table, but noting they are deprecated.  

It was decided to address this in 4.1.

It would also be great to come up with a way to auto generate this list.

# References and Pull Requests

**Comment by cniethammer** [comment]
A note here: The MPI_T_* functions are not included in the list here but mentioned in the text.
So a question for the discussion in this context: Leave this as is or include them as well for a comprehensive list?

**Comment by jsquyres** [comment]
> A note here: The MPI_T_* functions are not included in the list here but mentioned in the text.
> So a question for the discussion in this context: Leave this as is or include them as well for a comprehensive list?

Fair point.  Should we just say that the table is not comprehensive/complete?

**Comment by cniethammer** [comment]
Instead of a central table we could also display this information as part of the LIS. Personally I think it could be very convenient for a user to have this information close to the function itself.

**Comment by schulzm** [comment]
Comment on 7/21 by @hppritcha : similar to issue #497

**Comment by hppritcha** [comment]
@martinruefenacht would you be able to work on this for 4.1?  Otherwise i'm inclined to put it off to 5.0.

**Comment by hppritcha** [comment]
moving to 5.0
